cmd/vela-makisu/push.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2020 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package main
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+const pushAction = "push"
+
+type (
+	// Push represents the plugin configuration for push information.
+	//
+	// Makisu documents their command usage:
+	// https://githup.com/uber/makisu/blob/master/docs/COMMAND.md
+	// nolint
+	Push struct {
+		// enables setting the global flags
+		GlobalFlags []string
+		// enables setting the location of the tar file
+		Path string
+		// enables setting registries to push an image to
+		Pushes []string
+		// enables setting the credentials for authentication
+		RegistryConfig string
+		// enables pushing images with alternative full names
+		Replicas []string
+		// enables setting the tag for an image
+		Tag string
+	}
+)
+
+// pushFlags represents for config settings on the cli.
+// nolint
+var pushFlags = []cli.Flag{
+	&cli.StringFlag{
+		EnvVars:  []string{"PARAMETER_PATH"},
+		FilePath: string("/vela/parameters/makisu/push/path,/vela/secrets/makisu/push/path"),
+		Name:     "push.path",
+		Usage:    "enables setting the location of the tar file",
+	},
+	&cli.StringSliceFlag{
+		EnvVars:  []string{"PARAMETER_PUSHES"},
+		FilePath: string("/vela/parameters/makisu/push/context,/vela/secrets/makisu/push/context"),
+		Name:     "push.pushes",
+		Usage:    "enables setting registries to push an image to",
+	},
+	&cli.StringSliceFlag{
+		EnvVars:  []string{"PARAMETER_REPLICAS"},
+		FilePath: string("/vela/parameters/makisu/push/context,/vela/secrets/makisu/push/context"),
```

Comment:
Are the paths here accurate?

I wonder if they could be named a bit closer to the flag and env var naming 👍

```suggestion
		FilePath: string("/vela/parameters/makisu/push/replicas,/vela/secrets/makisu/push/replicas"),
```